src/features/user-dashboard/HomeRoutes.tsx

```diff
@@ -0,0 +1,32 @@
+import { Layout } from 'common/components/Layout';
+import { NotFoundView } from 'common/components/NotFoundView';
+import { useAuth } from 'features/auth/hooks';
+import { FC } from 'react';
+import { Route, Routes } from 'react-router-dom';
+import { AdminHomePage } from './pages/AdminHomePage';
+import { EditorHomePage } from './pages/EditorHomePage';
+import { UserHomePage } from './pages/UserHomePage';
+
+export const HomeRoutes: FC = () => {
+  const { user } = useAuth();
+
+  return (
+    <Layout>
+      <Routes>
+        <Route
+          path='/'
+          element={
+            user
```

Comment:
One of the problems I'd like to solve on our boilerplate frontend in general is that HomeRoutes is wrapped in `RequireAuth` and yet the type system is still unsure that we have a user or not.

We have to have a user if we are here so it adds extra checking that isn't necessary. I've been getting around it with `user!.role` telling the type system I know there is a user, however theres got to be a better way.

Not something you have to resolve for this PR, but something to think about. I would at least switch to `user!.role` instead so you don't have to do the check.